src/ui/box.js

```diff
@@ -36,18 +36,27 @@ class Renderer {
       CSSCore.addClass(window.document.getElementsByTagName('html')[0], 'auth0-lock-html');
     }
     // eslint-disable-next-line
-    const component = ReactDOM.render(<Container {...props} />, container);
+    const root = this.modals[containerId] ? this.modals[containerId].root : createRoot(container);
 
-    if (isModal) {
-      this.modals[containerId] = component;
+    if (!this.modals[containerId]) {
+      this.modals[containerId] = { root };
     }
 
-    return component;
+    root.render(
+      <Container
+        {...props}
+        ref={component => {
+          if (component && isModal) {
+            this.modals[containerId].component = component;
+          }
+        }}
+      />
+    );
   }
 
   remove(containerId) {
     if (this.modals[containerId]) {
-      this.modals[containerId].hide();
+      this.modals[containerId].component.hide();
```

Comment:
My app is crashing on this line.
On IOS (Mobile Safari UI/WKWebVie) / MAC (chrome 109) sometimes when unmount auth0-lock, i'm not sure how to reproduce it always so not sure I can open an issue about it.

I suggest maybe to optional call `hide()` ?

This: `this.modals?.[containerId]?.component?.hide?.();`
or this: `this.modals[containerId].component &&this.modals[containerId].component.hide();`

<img width="917" alt="Screen Shot 2023-01-26 at 13 34 36" src="https://user-images.githubusercontent.com/3075913/214825833-fcdaaea1-491e-4d41-ad7c-333318e0072d.png">